src/libp2p/peer_manager.rs

```diff
@@ -124,30 +124,49 @@ impl PeerManager {
     pub(in crate::libp2p) fn sorted_peers(&self) -> Vec<PeerId> {
         let peer_lk = self.peers.read();
         let average_time = self.avg_global_time.read();
+        let mut n_stateful = 0;
         let mut peers: Vec<_> = peer_lk
             .full_peers
             .iter()
-            .filter_map(|(p, info)| {
-                // Filter out nodes that are stateless (or far behind)
-                if info.head.epoch() > 0 {
-                    let cost = if info.successes > 0 {
-                        // Calculate cost based on fail rate and latency
-                        let fail_rate = f64::from(info.failures) / f64::from(info.successes);
-                        info.average_time.as_secs_f64() + fail_rate * average_time.as_secs_f64()
-                    } else {
-                        // There have been no failures or successes
-                        average_time.as_secs_f64() * NEW_PEER_MUL
-                    };
-                    Some((p, cost))
-                } else {
-                    None
+            .map(|(&p, info)| {
+                let is_stateful = info.head.epoch() > 0;
+                if is_stateful {
+                    n_stateful += 1;
                 }
+                let cost = if info.successes + info.failures > 0 {
```

Comment:
wibble: this was changed to `if info.successes > 0` to address https://github.com/ChainSafe/forest/pull/4376#discussion_r1636561639
However, this unexpectedly changes the logic to favor peers that never succeed.
The original code does not panic but returns `inf`

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=657686952703dcee856841d192b99774

```rust
fn main() {
    let x = 100.0 + f64::from(1) / f64::from(0);
    println!("{x}");
}

// Output: inf
```